web_external/js/models/MongoSourceModel.js

```diff
@@ -0,0 +1,23 @@
+minerva.models.MongoSourceModel = minerva.models.SourceModel.extend({
+
+    initialize: function () {
+    },
+
+    createSource: function (params) {
+        girder.restRequest({
+            path: '/minerva_source_mongo',
+            type: 'POST',
+            data: params,
+            error: null // don't do default error behavior (validation may fail)
+        }).done(_.bind(function (resp) {
+            this.set(resp);
+            this.trigger('m:sourceReceived');
+        }, this)).error(_.bind(function (err) {
+            this.trigger('m:error', err);
+            // If we couldn't successfully start the s3 import destroy the item.
+            this.destroy();
```

Comment:
Thanks, I removed the destroy call

> On Nov 20, 2015, at 9:44 PM, mgrauer <notifications@github.com> wrote:
> 
> In web_external/js/models/MongoSourceModel.js <https://github.com/Kitware/minerva/pull/206#discussion_r45540193>:
> 
> > +    initialize: function () {
> > +    },
> > +
> > +    createSource: function (params) {
> > +        girder.restRequest({
> > +            path: '/minerva_source_mongo',
> > +            type: 'POST',
> > +            data: params,
> > +            error: null // don't do default error behavior (validation may fail)
> > +        }).done(_.bind(function (resp) {
> > +            this.set(resp);
> > +            this.trigger('m:sourceReceived');
> > +        }, this)).error(_.bind(function (err) {
> > +            this.trigger('m:error', err);
> > +            // If we couldn't successfully start the s3 import destroy the item.
> > +            this.destroy();
> You don't need to do this explicit destroy call.
> 
> The S3 Source created an additional folder to store the contents of the files imported from S3, and the destroy call of the Girder Item needed to be overridden so that the additional folder was deleted if there was an error creating the S3 source.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/Kitware/minerva/pull/206/files#r45540193>.
>